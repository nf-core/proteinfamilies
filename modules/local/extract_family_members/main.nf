process EXTRACT_FAMILY_MEMBERS {
    tag "$meta.id"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/7d/7d0fee0217685dc2501570e1dd12076f9466175d0a37335ca424390cffec5fa1/data' :
        'community.wave.seqera.io/library/python:3.13.1--d00663700fcc8bcf' }"

    input:
    tuple val(meta), path(faa, stageAs: "faa/*")

    output:
    tuple val(meta), path("${prefix}.tsv"), emit: tsv
    path "versions.yml"                   , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    extract_family_members.py \\
        --fasta_folder faa \\
        --num_threads ${task.cpus} \\
        --out_tsv ${prefix}.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
    END_VERSIONS
    """
}
