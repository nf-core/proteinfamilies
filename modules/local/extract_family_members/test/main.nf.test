nextflow_process {

    name "Test Process EXTRACT_FAMILY_MEMBERS"
    script "../main.nf"
    process "EXTRACT_FAMILY_MEMBERS"

    test("proteinfamilies - faa") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'extract_family_members' ],
                    [
                        file(params.pipelines_testdata_base_path + 'test_data/modules/mgnifams_test_1.faa', checkIfExists: true),
                        file(params.pipelines_testdata_base_path + 'test_data/modules/mgnifams_test_2.faa', checkIfExists: true)
                    ]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.tsv,
                    process.out.versions.collect { path(it).yaml }
                    ).match()}
            )
        }
    }

    test("proteinfamilies - faa - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'extract_family_members' ],
                    [
                        file(params.pipelines_testdata_base_path + 'test_data/modules/mgnifams_test_1.faa', checkIfExists: true),
                        file(params.pipelines_testdata_base_path + 'test_data/modules/mgnifams_test_2.faa', checkIfExists: true)
                    ]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
