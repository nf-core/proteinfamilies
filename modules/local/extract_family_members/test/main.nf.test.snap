{
    "proteinfamilies - faa - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "extract_family_members"
                        },
                        "extract_family_members.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,a65f813f91dfee556b11f53c62863d86"
                ],
                "tsv": [
                    [
                        {
                            "id": "extract_family_members"
                        },
                        "extract_family_members.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a65f813f91dfee556b11f53c62863d86"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-29T09:36:10.229298159"
    },
    "proteinfamilies - faa": {
        "content": [
            [
                [
                    {
                        "id": "extract_family_members"
                    },
                    "extract_family_members.tsv:md5,520d0a3dec8439ac4544b34d52bd15b3"
                ]
            ],
            [
                {
                    "EXTRACT_FAMILY_MEMBERS": {
                        "python": "3.13.1"
                    }
                }
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-29T09:36:01.267716479"
    }
}